liferay-headless/liferay-portal
# LPS-164967 Inject resources into templates

With these changes we can inject headless resources into templates. 

```
${resourceLocator.locate("/headless-delivery/v1.0", "BlogPosting").getBlogPosting(44505)}
```

To test it:

1. Go to `Design -> Templates -> Widget Templates`
2. Create a `Blogs Template`.
3. Add the Blogs Widget to the page.
4. In the configuration of the widget, under `Display Template`, select the template created in step 2.
5. Create a blog entry and copy its id.
6. Edit the template from step 2 and add this:
```
${resourceLocator.locate("/headless-delivery/v1.0", "BlogPosting").getBlogPosting(<the id from step 5>)}
```
7. Render the homepage and you should see the json:
<img width="850" alt="Screenshot 2022-10-07 at 14 31 22" src="https://user-images.githubusercontent.com/1937354/194553868-706f527d-8e3e-4390-a30a-ffed374b2616.png">

**Pending things:**
1. Objects support. We will create a separate ticket for it and plan it for the future.
